src/main/scala/lc/Main.scala

```diff
@@ -138,13 +137,17 @@ class Captcha(throttle: Int, dbConn: DBConn) {
   def getChallenge(param: Parameters): Id = {
     try {
       val tokenPstmt = Statements.tlStmts.get.tokenPstmt
+      val updateTimestampPstmt = Statements.tlStmts.get.updateTimestampPstmt
       val rs = tokenPstmt.executeQuery()
       val tokenOpt = if(rs.next()) {
         Some(rs.getInt("token"))
       } else {
         None
       }
-      Id(getUUID(tokenOpt.getOrElse(generateChallenge(param))))
+      val uuid = getUUID(tokenOpt.getOrElse(generateChallenge(param)))
+      updateTimestampPstmt.setString(1, uuid)
+      updateTimestampPstmt.executeUpdate()
```

Comment:
Since we are generating a unique UUID everytime, can we set the timestamp while creating the map entry? Then we won't need to make a separate database update.

Correct me if I am wrong. Haven't looked at the details fully.

Reply:
Yeah, that will work.